src/js/pagination.js

```diff
@@ -0,0 +1,109 @@
+document.addEventListener("DOMContentLoaded", async () => {
+  setupPagination(fetchAndRenderExercises);
+});
```

Comment:
move it after you have declared function fetchAndRenderExercises

Reply:
done